src/Symfony/Bridge/Doctrine/ManagerRegistry.php

```diff
@@ -81,15 +81,15 @@ function (&$wrappedInstance, LazyLoadingInterface $manager) use ($name) {
         }
 
         try {
-            $r->resetAsLazyProxy($manager, \Closure::bind(
-                function () use ($name) {
+            $r->resetAsLazyGhost($manager, \Closure::bind(
```

Comment:
Let's make the code even more robust: the manager could be a proxy depending on the way the service is configured.
To decide, we need to call $r->initializeLazyObject: if this returns the same object, it's a ghost, otherwise a proxy.
Then I think we can make the closure's code generic with a `$asGhost =$r->initializeLazyObject($manager) === $manager;` to provide the correct behavior.